From 6556cad99dd9f017d7aadc96faa2c1975cba555f Mon Sep 17 00:00:00 2001
From: Stijn Tintel <stijn@linux-ipv6.be>
Date: Tue, 28 Jun 2022 22:14:29 +0300
Subject: [PATCH] hostapd: disable mbo by default

Enabling mbo by default on 802.11ax devices breaks for encryption types
that do not enable 802.11w by default. Disable mbo by default to fix
this. Enabling mbo by default on 802.11ax devices was not explained in
the commit message anyway.

Fixes: 6eee9836565c ("hostapd: introduce mbo option")
Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
---
 package/network/services/hostapd/files/hostapd.sh | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/package/network/services/hostapd/files/hostapd.sh b/package/network/services/hostapd/files/hostapd.sh
index be3993d4bd..760b3ff792 100644
--- a/package/network/services/hostapd/files/hostapd.sh
+++ b/package/network/services/hostapd/files/hostapd.sh
@@ -829,12 +829,7 @@ hostapd_set_bss_options() {
 	set_default bss_transition 0
 	set_default wnm_sleep_mode 0
 	set_default wnm_sleep_mode_no_keys 0
-
-	if [ "$enable_ax" -eq 1 ]; then
-		set_default mbo 1
-	else
-		set_default mbo 0
-	fi
+	set_default mbo 0
 
 	[ -n "$time_advertisement" ] && append bss_conf "time_advertisement=$time_advertisement" "$N"
 	[ -n "$time_zone" ] && append bss_conf "time_zone=$time_zone" "$N"
-- 
2.35.1

